src/Discovery/src/Consul/Steeltoe.Discovery.Consul.csproj

```diff
@@ -18,5 +18,6 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\..\Common\src\Http\Steeltoe.Common.Http.csproj" />
+    <ProjectReference Include="..\..\..\Configuration\src\CloudFoundry\Steeltoe.Configuration.CloudFoundry.csproj" />
```

Comment:
Can/should this be a private reference so that it isn't a hard dependency?

Reply:
I suppose it can, but I'm not a huge fan of all those magical references. And to test it properly, we'd need two test projects: one with and one without the reference. We do that for the dependency from Config Server to Eureka/Consul because they're very different systems. But Cloud Foundry is so integrated already that I believe having an explicit reference makes dependencies just easier to understand. We already have `Platform.IsCloudFoundry` in Common that's being referenced, so the dependencies on Cloud Foundry are quite blurry.

On the other hand, having an explicit reference indirectly pulls in
```xml
<FrameworkReference Include="Microsoft.AspNetCore.App" />
```
so that may be a reason for concern. Should we support using Consul/Eureka from a console app?

I'm not sure what to do here, do you have a strong preference?

Reply:
> On the other hand, having an explicit reference indirectly pulls in

Oh, that's not true. The dependency is needed anyway.

Reply:
Agree on blurry lines, maybe it isn't a big deal as long as the things we use everywhere stay extremely lightweight... 
`Should we support using Consul/Eureka from a console app?` - ASP.NET Core should probably _not_ be a requirement in order to discover services. I don't remember that changing, but it does look like we already have the framework ref in Consul/Eureka

Reply:
I remember testing Eureka in a console app. While the reference is there, the consuming code uses `serviceProvider.GetService` instead of `GetRequiredService` to cope with that.